pydantic_ai_slim/pydantic_ai/models/openrouter.py

```diff
@@ -0,0 +1,637 @@
+    """Represents the 'Provider' object from the OpenRouter API."""
+
+    order: list[OpenRouterProviderName]
+    """List of provider slugs to try in order (e.g. ["anthropic", "openai"]). [See details](https://openrouter.ai/docs/features/provider-routing#ordering-specific-providers)"""
+
+    allow_fallbacks: bool
+    """Whether to allow backup providers when the primary is unavailable. [See details](https://openrouter.ai/docs/features/provider-routing#disabling-fallbacks)"""
+
+    require_parameters: bool
+    """Only use providers that support all parameters in your request."""
+
+    data_collection: Literal['allow', 'deny']
+    """Control whether to use providers that may store data. [See details](https://openrouter.ai/docs/features/provider-routing#requiring-providers-to-comply-with-data-policies)"""
+
+    zdr: bool
+    """Restrict routing to only ZDR (Zero Data Retention) endpoints. [See details](https://openrouter.ai/docs/features/provider-routing#zero-data-retention-enforcement)"""
+
+    only: list[OpenRouterProviderName]
+    """List of provider slugs to allow for this request. [See details](https://openrouter.ai/docs/features/provider-routing#allowing-only-specific-providers)"""
+
+    ignore: list[str]
+    """List of provider slugs to skip for this request. [See details](https://openrouter.ai/docs/features/provider-routing#ignoring-providers)"""
+
+    quantizations: list[Literal['int4', 'int8', 'fp4', 'fp6', 'fp8', 'fp16', 'bf16', 'fp32', 'unknown']]
+    """List of quantization levels to filter by (e.g. ["int4", "int8"]). [See details](https://openrouter.ai/docs/features/provider-routing#quantization)"""
+
+    sort: Literal['price', 'throughput', 'latency']
+    """Sort providers by price or throughput. (e.g. "price" or "throughput"). [See details](https://openrouter.ai/docs/features/provider-routing#provider-sorting)"""
+
+    max_price: OpenRouterMaxPrice
+    """The maximum pricing you want to pay for this request. [See details](https://openrouter.ai/docs/features/provider-routing#max-price)"""
+
+
+class OpenRouterReasoning(TypedDict, total=False):
+    """Configuration for reasoning tokens in OpenRouter requests.
+
+    Reasoning tokens allow models to show their step-by-step thinking process.
+    You can configure this using either OpenAI-style effort levels or Anthropic-style
+    token limits, but not both simultaneously.
+    """
+
+    effort: Literal['high', 'medium', 'low']
+    """OpenAI-style reasoning effort level. Cannot be used with max_tokens."""
+
+    max_tokens: int
+    """Anthropic-style specific token limit for reasoning. Cannot be used with effort."""
+
+    exclude: bool
+    """Whether to exclude reasoning tokens from the response. Default is False. All models support this."""
+
+    enabled: bool
+    """Whether to enable reasoning with default parameters. Default is inferred from effort or max_tokens."""
+
+
+class WebPlugin(TypedDict, total=False):
```

Comment:
Could/should this be supported via the `WebSearchTool` built-in tool? https://ai.pydantic.dev/builtin-tools/#web-search-tool